### DISCUSSION TITLE
Financial Spending Plan - Request for Comments

### DISCUSSION THREAD

**Discussion by Flyingmana** [OpenMage/magento-lts#3678] [General]
With OpenMage now accepting donations on OpenCollective, we should make some plans on what we want to spend money on and at which order, so possible sponsors get an idea what we will use the money for.

I already created a rough plan of things I thought make sense. https://github.com/OpenMage/organizational/blob/main/finance/expense_planning.md

But what do you think about it? Would you add things? Would you put a different priority or grouping?
What are your thoughts about it?

**Comment by addison74** [comment]
OpenMage needs a permanent team to maintain the source code. I hope that this money that will be collected will be invested so that OpenMage can go further.

Personally, I would like to see that all PR and Issues reported are analyzed and solved in a time range. Then let's get rid of Prototype, parts of outdated PHP code. A new professional theme for Frontend and one for Backend. Several integrated "must-have" features (management of buyer fields, address, ...) posted in Discussions > Ideas.

**Reply by addison74** [reply]
... and maybe we will end up replacing ZF1-Future with Laminas.

**Comment by kiatng** [comment]
As a way to market the product, OpenMage, Level 1 should have some budget for updating the site https://www.openmage.org/. This can create a positive feedback loop: marketing -> adoption -> sponsorship ->sustainability

**Reply by sreichel** [reply]
The last thing i would spent money for is that webpage. @colinmollenhour friend/co-worker did a good job with this, but it got never really finished. Code highlighting in texts looks not so nice. There are no content updates. Incomplete install instructions. Partner page got no updates for monthes - but there are open requests. @Flyingmana ?) 

I'd strip it down to a minimum - that need no frequent updates (remove releases, install, ...) and forward to repo documentation.

**Reply by kiatng** [reply]
Successful open source projects that I know of all have nice public facing website (for both technical and non-technical audience) to drive adoption. Very often, decision makers are non-technical people. To quote an internal 3M sales training I received years ago: Nothing happens until someone sells something. As an engineer, it took me very long time to get it. For sustainability, we need sales and marketing. An updated website can be a good salesman, it deserves some budget.

**Reply by sreichel** [reply]
As said, we have a nice website. But for e.g. a not updated release-section does make it look unprofessional or unmaintained.

Is here anyone who wants to be "content-manager"? From my docu-PR i know the time it takes, to add and refactor content. This can be a full-time job ... 

Have a nice looking homepage, yeah. Spent money for updates no, not yet.

**Reply by fballiano** [reply]
> This can be a full-time job

that's why it should have some budget ;-)

**Reply by fballiano** [reply]
btw if mkdocs became the website (as many projects do nowadays) it would be easier for everybody.

**Comment by sreichel** [comment]
IMHO ... money should go to make code more solid.

- fix baselined baselined phpstan errors
- fix phpstan errors up to L6+ to ensure code quality (7/8 is possible)
- fix doc blocks in template files (to get recognized in IDE)
- add strict_types (for cleaner code)
- turn method annotations into real getter/setter for all DB-related fields (add interfaces for that)
- **add unit tests**
- refactor (rector) code incl. rules to "patch" 3rd party code

Further:

- head over to zf1s, and only install Zend-parts that are required (makes it easier to replace them with others)
- modulize OM (already done in private repo ... later... )
- ...

**Comment by sreichel** [comment]
> > This can be a full-time job
> 
> that's why it should have some budget ;-)

For keeping docs/website up2date? Nah ...

Having docs inside repo (mkdocs is just easy to manage for nice results) maybe makes it easier to add or improve some content.

**Reply by fballiano** [reply]
You said it’s a full time job

**Reply by sreichel** [reply]
Does OM need a payed "content manager"? Anyone who wants to be?

Better give you the money for releases and taking care. Not for that homepage. (edit: give some payback to collins friend ;))

**Reply by fballiano** [reply]
I agree and that’s why i said many months ago that website should be a redirect to the repo, but then it’s not a full time job as you said

**Reply by sreichel** [reply]
Got me wrong ... it is/can be a full time job to have an frequently updated homepage.

**Reply by toejam34** [reply]
I feel this needs to be revisited now.

Whilst I appreciate I have done little for the community as a whole ( or even a part! ) as a consumer I see the homepage is inconsistent with its information and some is out of date. 

I would dearly like to offer my skills to be able to update the site but I have a full time job so doing so would not be feasible. 

I like @fballiano idea of redirecting it to the github. At least the data is consistent then

**Comment by sreichel** [comment]
@Flyingmana can you provide sample output of https://github.com/OpenMage/organizational/blob/main/changelog/generate_base_data.php?

Possible to split it into `Mage_Xyz module`?

**Comment by ysrtech** [comment]
I suggest we build a M2 API emulator as many popular services offer only M2 API integration. Many companies were forced to moved to M2 just because if this.

**Reply by pquerner** [reply]
Can you give more details for this?

**Reply by ysrtech** [reply]
There are many popular services that offer only M2 integration but not M1. One quick example. https://github.com/n8n-io/n8n
I actually built my own M2 API emulator to work with a local shipping aggregator, but it still needs some work and I believe many current Open Mage users will be forced at some point to move to M2 or Shopify for this reason

**Reply by pquerner** [reply]
I still dont get it. :D 
When you say M2 API Emulator, what are you emulating there? The routes to be matching or other services to connect to your Magento installation? Pretending to be M2 so that their tools work on your instance?

**Reply by ysrtech** [reply]
Exactly, its pretending to be M2 ;-).  I've managed to pretend being M2 for order processing and shipments, but its all hacks, needs a real module.

**Reply by pquerner** [reply]
Mh. Good idea. But I know too little about M2 to be doing that, really. I dont know the quirks, I dont know the output format. And it will be needed to keep up to date with the latest M2 for all eternity. Very difficult.

**Reply by ysrtech** [reply]
I'm also not an expert with M2, but I've worked with talented devs who know M1 and M2 inside out.

**Reply by Flyingmana** [reply]
Having an API endpoint which simulates the same input/output format as M2, is surely useful.
And thanks to swagger and similar API documentation, its easy enough to see how the API of M2 looks: https://developer.adobe.com/commerce/webapi/rest/quick-reference/

We are probably not able to do this for all of them, but the ones usually needed should be doable.
For the Beginning we could start it as separate module, or maybe someone has already a module linked we could extend together?

Of course this is also a project which we could finance when having enough funding, in case no volunteers can be found.
